### ISSUE TITLE
MURA 7 + CF 11, Core and Site Version 7.0.6879, data type issue with MS SQL when saving events

### ISSUE THREAD

**Issue by CFJEFF** [blueriver/MuraCMS#2495]
500 Error
Error Executing Database Query. 

SQL: INSERT INTO tcontent (ContentHistID, ContentID, Active, OrderNo, approved, DisplayStart, displaystop, MetaDesc, MetaKeyWords, Body, Summary, path, tags, responseMessage, responseDisplayFields, Title, menuTitle, filename, LastUpdate, Display, ParentID, Type, subType,LastUpdateBy, LastUpdateByID, siteid, moduleid, isnav, restricted, target, restrictgroups,template, responseChart,responseSendTo,moduleAssign,notes,inheritObjects,isFeature, releaseDate,targetParams,isLocked,nextN,sortBy,sortDirection,featureStart,featureStop,fileID,forceSSL, remoteID,remoteURL,remotePubDate,remoteSource,RemoteSourceURL,Credits, audience, keyPoints, searchExclude, displayTitle, doCache, created, urltitle, htmltitle, mobileExclude, changesetID, imageSize, imageHeight, imageWidth, childTemplate, majorVersion, minorVersion, expires, displayInterval ,objectParams) VALUES ( (param 1) , (param 2) , 1, 1, 1, (param 3) , (param 4) , (param 5) , (param 6) , (param 7) , (param 8) , (param 9) , (param 10) , (param 11) , (param 12) , (param 13) , (param 14) , (param 15) , (param 16) , 2, (param 17) , (param 18) , (param 19) , (param 20) , (param 21) , (param 22) , (param 23) , 1, 0, (param 24) , (param 25) , (param 26) , 0, (param 27) , (param 28) , (param 29) , (param 30) , 0, null , Null , 0, (param 31) , (param 32) , (param 33) , null , null , (param 34) , (param 35) , (param 36) , (param 37) , (param 38) , (param 39) , (param 40) , (param 41) , (param 42) , (param 43) , 0, 0, 1, (param 44) , (param 45) , (param 46) , 0, (param 47) , (param 48) , (param 49) , (param 50) , (param 51) , 0, 0, null , (param 52) , (param 53) ) 

Code: 22007 

Type: Database 

[Macromedia][SQLServer JDBC Driver][SQLServer]The conversion of a datetimeoffset data type to a smalldatetime data type resulted in an out-of-range value. 

coldfusion.tagext.sql.QueryTag$DatabaseQueryException: Error Executing Database Query.
	at coldfusion.tagext.sql.QueryTag.startQueryExecution(QueryTag.java:867)
	at coldfusion.tagext.sql.QueryTag.doEndTag(QueryTag.java:769)
	at cfcontentDAO2ecfc425782685$funcCREATE.runFunction(E:\webs\internet\requirements\mura\content\contentDAO.cfc:705)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)
     etc, etc, etc,,,,,

Column: 0
ID: CFQUERY
Line: 705
Raw Trace: at cfcontentDAO2ecfc425782685$funcCREATE.runFunction(E:\webs\internet\requirements\mura\content\contentDAO.cfc:705)
Template: E:\webs\internet\requirements\mura\content\contentDAO.cfc
Type: CFML

**Comment by CFJEFF** [comment]
I can not reproduce this error on a clean install.   I have 2 servers that I upgraded from 6.2. 6517 to 6879.   These 2 upgraded servers are having the issue.   I did the upgrade through the admin and I didn't catch any errors.

**Comment by CFJEFF** [comment]
a comparison of the tcontent table shows the columns have a datetime data type on the working db and a smalldatetime datatype on the failing db.

- [lastUpdate]
- [DisplayStart]
- [DisplayStop]
- [ReleaseDate]
- [featureStart]
- [featureStop]

**Comment by mattlevine** [comment]
Can you manually update those columns to datetime instead of smalldatetime and reload and retest.

**Comment by CFJEFF** [comment]
I tried doing an alter these fields in the tcontent but there were some interdependencies with other tables.

**Comment by mattlevine** [comment]
I'm not aware of any interdependencies.  There are no joins on those columns or anything like that?

**Comment by CFJEFF** [comment]
use w3MuraCms
go

alter table tcontent
alter column lastUpdate datetime

go

alter table tcontent
alter column DisplayStart datetime

go


alter table tcontent
alter column DisplayStop datetime

go


alter table tcontent
alter column ReleaseDate datetime

go


alter table tcontent
alter column featureStart datetime

go
alter table tcontent
alter column featureStop datetime

go

results in:  

Msg 5074, Level 16, State 1, Line 4
The index 'IX_tcontent_displaystart' is dependent on column 'DisplayStart'.
Msg 5074, Level 16, State 1, Line 4
The index 'ontent_SiteID_Active_Approved_IsNav_ModuleID_searchExclude_ContentID_Type_subType_Display_DisplayStart_DisplayStop_mobileExclude' is dependent on column 'DisplayStart'.
Msg 4922, Level 16, State 9, Line 4
ALTER TABLE ALTER COLUMN DisplayStart failed because one or more objects access this column.
Msg 5074, Level 16, State 1, Line 10
The index 'IX_tcontent_displaystop' is dependent on column 'DisplayStop'.
Msg 5074, Level 16, State 1, Line 10
The index 'ontent_SiteID_Active_Approved_IsNav_ModuleID_searchExclude_ContentID_Type_subType_Display_DisplayStart_DisplayStop_mobileExclude' is dependent on column 'DisplayStop'.
Msg 4922, Level 16, State 9, Line 10
ALTER TABLE ALTER COLUMN DisplayStop failed because one or more objects access this column.

**Comment by mattlevine** [comment]
I would drop those indexes and then after updating the datatypes do an appreload&applydbupdates and Mura will add them back.

**Comment by CFJEFF** [comment]
I'll try that later today.

**Comment by CFJEFF** [comment]
This appears to have worked.   After running the applydbuappreload&applydbupdates I still do not see the indexes I dropped  so I'm not clear that the indexes where added back.  (I checked using the gui in the sql management studio).   It doesn't appear the mssql.sql in the setup creates these indexes.

I did run across an issue with an event for one day showing up forever until I checked and unchecked the "repeats" checkbox.

Also having a weird problem where MURA fails to start unless I uncheck the "Disable access to internal ColdFusion Java components" in the CF Admin.    Not sure if that's related to anything.   I posted this to google groups.
